lib/danger/danger_core/plugins/dangerfile_bitbucket_server_plugin.rb

```diff
@@ -0,0 +1,179 @@
+      "bitbucket_server"
+    end
+
+    def initialize(dangerfile)
+      super(dangerfile)
+      @bs = dangerfile.env.request_source
+    end
+
+    # @!group Bitbucket Server Misc
+    # The hash that represents the PR's JSON. For an example of what this looks like
+    # see the [Danger Fixture'd one](https://raw.githubusercontent.com/danger/danger/master/spec/fixtures/bitbucket_server_api/pr_response.json).
+    # @return [Hash]
+    def pr_json
+      @bs.pr_json
+    end
+
+    # @!group PR Metadata
+    # The title of the Pull Request.
+    # @return [String]
+    #
+    def pr_title
+      @bs.pr_json[:title].to_s
+    end
+
+    # @!group PR Metadata
+    # The body text of the Pull Request.
+    # @return [String]
+    #
+    def pr_description
+      @bs.pr_json[:description].to_s
+    end
+    alias pr_body pr_description
+
+    # @!group PR Metadata
+    # The username of the author of the Pull Request.
+    # @return [String]
+    #
+    def pr_author
+      @bs.pr_json[:author][:user][:slug].to_s
+    end
+
+    # @!group PR Commit Metadata
+    # The branch to which the PR is going to be merged into.
+    # @return [String]
+    #
+    def branch_for_base
+      @bs.pr_json[:toRef][:displayId].to_s
+    end
+
+    # @!group PR Commit Metadata
+    # A href that represents the current PR
+    # @return [String]
+    #
+    def pr_link
+      @bs.pr_json[:links][:self].flat_map { |l| l[:href] }.first.to_s
+    end
+
+    # @!group PR Commit Metadata
+    # The branch to which the PR is going to be merged from.
+    # @return [String]
+    #
+    def branch_for_head
+      @bs.pr_json[:fromRef][:displayId].to_s
+    end
+
+    # @!group PR Commit Metadata
+    # The base commit to which the PR is going to be merged as a parent.
+    # @return [String]
+    #
+    def base_commit
+      @bs.pr_json[:toRef][:latestCommit].to_s
+    end
+
+    # @!group PR Commit Metadata
+    # The head commit to which the PR is requesting to be merged from.
+    # @return [String]
+    #
+    def head_commit
+      @bs.pr_json[:fromRef][:latestCommit].to_s
+    end
+
+    # @!group Bitbucket Server Misc
+    # Returns a list of HTML anchors for a file, or files in the head repository.
+    # It returns a string of multiple anchors if passed an array.
+    # @param    [String or Array<String>] paths
+    #           A list of strings to convert to github anchors
+    # @param    [Bool] full_path
+    #           Shows the full path as the link's text, defaults to `true`.
+    #
+    # @return [String]
+    #
+    def html_link(paths, full_path: true)
+      paths = [paths] unless paths.kind_of?(Array)
+      commit = head_commit
+      repo = pr_json[:fromRef][:repository][:project][:links][:self].flat_map { |l| l[:href] }.first
+
+      paths = paths.map do |path|
+        url_path = path.start_with?("/") ? path : "/#{path}"
+        text = full_path ? path : File.basename(path)
+        create_link("#{repo}/blob/#{commit}#{url_path}", text)
```

Comment:
@HeEAaD can you please confirm what a file path would look like? 

For example on the bitbucket site it looks like 
`https://bitbucket.org/atlassian/bitbucket-example-plugin/src/a55cffe7744e8f50abcd2f092243b4f7e1d855b1/README.md?at=master&fileviewer=file-view-default`

GitHub's looks like: `https://github.com/danger/danger/blob/199c4b61925685db1e79569a2d34450c73ece089/lib/danger/comment_generators/bitbucket_server.md.erb`